.github/workflows/validations.yml

```
@@ -6,98 +6,329 @@ on:
     if: "!contains(github.event.head_commit.message, 'skip ci')"
+    strategy:
+      fail-fast: false
+      matrix:
+        # Run tests in 4 parallel jobs for better distribution
+        ci_node_index: [0, 1, 2, 3]
+        ci_node_total: [4]
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
+      
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)